### PR TITLE
Implement multi selection

### PR DESCRIPTION
**Pull request by psarram**
- Modify the Selection API to support selecting multiple objects via shift click. Shift click adds and remove from selection. 
- Implement multi-move and multi-resize
- Modify toolbar to support multi-select
- Remove snap to grid feature
- Modify the migration script to convert position and dimensions of objects with snap to grid disabled to grid.

### CODE REVIEW

File: src/selection/Selection.js [hunk 1]
```diff
@@ -47,38 +55,124 @@ define(['EventEmitter'], function (EventEmitter) {
...
+    Selection.prototype.select = function (selectable, isMultiSelectEvent) {
         if (!Array.isArray(selectable)) {
             selectable = [selectable];
         }
 
-        if (this.selected[0] && this.selected[0].element) {
-            this.selected[0].element.removeAttribute('s-selected');
+        let multiSelect = isMultiSelectEvent &&
+            this.parentSupportsMultiSelect(selectable) &&
+            this.peerSelection(selectable) &&
```
Comment (akhenry) [inline src/selection/Selection.js]: It took me a moment to figure out what this function did. Could we name it something like `isPeer()`?

File: src/ui/inspector/Inspector.vue [hunk 2]
```diff
@@ -199,8 +199,8 @@
         },
         methods: {
             refreshComposition(selection) {
-                if (selection[0]) {
-                    let parentObject = selection[0].context.item;
+                if (selection && selection[0]) {
```
Comment (akhenry) [inline src/ui/inspector/Inspector.vue]: Do we need to check if `selection` is defined?

File: src/adapter/views/TypeInspectorViewProvider.js [hunk 3]
```diff
@@ -25,14 +25,21 @@ define([
...
             canView: function (selection) {
-                if (!selection[0] || !selection[0].context.item) {
+                if (selection.length === 0 || selection[0].length === 0) {
                     return false;
                 }
-                let domainObject = selection[0].context.item;
+
+                let selectionPath = selection[0];
+
+                if (!selectionPath[0] || !selectionPath[0].context.item) {
```
[the remark this answers is not in the document]
Comment (psarram) [reply src/adapter/views/TypeInspectorViewProvider.js]: selection can be an empty array which is checked at line 28 but not undefined. Checking for selectionPath[0] being undefined is unnecessary. We only need to check for item on context.

File: src/selection/Selection.js (hunk 1 above)
Comment (psarram) [reply src/selection/Selection.js]: You're right, isPeer() is a better name and I renamed the function.

File: src/ui/inspector/Inspector.vue (hunk 2 above)
Comment (psarram) [reply src/ui/inspector/Inspector.vue]: Good catch, selection is always defined as an array. I removed that check.

**Comment by psarram** [comment]
@akhenry Thanks for the review! I addressed the 3 suggested changes. I will make a note of the other 2 items to address later.

**Comment by psarram** [comment]
@akhenry I noticed the conflicts in the branch. Will resolve it soon.

File: src/ui/inspector/Inspector.vue (hunk 2 above)
Comment (psarram) [reply src/ui/inspector/Inspector.vue]: Actually, we need to check for selection and selection[0] length be greater than 0.

**Comment by akhenry** [comment]
@psarram Changes look good, thanks. There is a merge conflict (probably due to other PRs merged today)

**Comment by psarram** [comment]
@akhenry resolved the latest conflicts.

**Pull request merged**